components/scream/src/physics/share/physics_universal_impl.hpp

```diff
@@ -106,7 +106,30 @@ Functions<S,D>::get_dz(const Spack& zi_top, const Spack& zi_bot, const Smask& ra
   return result;
 }
 //-----------------------------------------------------------------------------------------------//
+// Compute dry static energy (DSE).
+// The result unit is in J/kg
+// The inputs are
+//   T_mid is the atmospheric temperature. Units in K.
+//   z_mid is the geopotential height above surface at midpoints. Units in m.
+//   surf_geopotential is the surface geopotential height. Units in m.
+template <typename S, typename D>
+KOKKOS_FUNCTION
+typename Functions<S,D>::Spack
+Functions<S,D>::get_dse(const Spack& T_mid, const Spack& z_mid, const Real surf_geopotential, const Smask& range_mask)
+{
+  static constexpr Scalar cp  = C::CP;
+  static constexpr Scalar ggr = C::gravit;
 
+  Spack result;
+  const Spack dse = cp*T_mid + ggr*z_mid + surf_geopotential;
+  // Check that there are no obvious errors in the result.
+  EKAT_KERNEL_ASSERT_MSG(!((isnan(dse) && range_mask).any()), "Error in get_dse, dse has NaN values.\n"); // exit with an error message
```

Comment:
I think it makes more sense to check the inputs for any nan, no? If this throws, we still don't know what caused it. We might instead check T_mid, z_mid, and the surf geopot, and print exactly which var is nan.

This is only done in dbg mode anyways.